concrete/testtool/testtool.go

```diff
@@ -43,84 +42,81 @@ var (
 	PrintLogs = true
 )
 
-func runTestMethod(concreteRegistry concrete.PrecompileRegistry, bytecode []byte, method abi.Method, shouldFail bool) (uint64, error) {
-	var (
-		key, _          = crypto.HexToECDSA("d17bd946feb884d463d58fb702b94dd0457ca349338da1d732a57856cf777ccd") // 0xCcca11AbAC28D9b6FceD3a9CA73C434f6b33B215
-		senderAddress   = crypto.PubkeyToAddress(key.PublicKey)
-		contractAddress = common.HexToAddress("cc73570000000000000000000000000000000000")
-		gspec           = &core.Genesis{
+func runTestMethod(t *testing.T, concreteRegistry concrete.PrecompileRegistry, bytecode []byte, method abi.Method, shouldFail bool) {
+	t.Run("runTestMethod", func(t *testing.T) {
+		key, _ := crypto.HexToECDSA("d17bd946feb884d463d58fb702b94dd0457ca349338da1d732a57856cf777ccd") // 0xCcca11AbAC28D9b6FceD3a9CA73C434f6b33B215
+		senderAddress := crypto.PubkeyToAddress(key.PublicKey)
+		contractAddress := common.HexToAddress("cc73570000000000000000000000000000000000")
+		gspec := &core.Genesis{
 			GasLimit: 2e9,
 			Config:   params.TestChainConfig,
 			Alloc: types.GenesisAlloc{
 				senderAddress:   {Balance: big.NewInt(1e18)},
 				contractAddress: {Balance: common.Big0, Code: bytecode},
 			},
 		}
-		signer   = types.LatestSigner(gspec.Config)
-		gasLimit = uint64(1e7)
-		setupId  = crypto.Keccak256([]byte("setUp()"))[:4]
-	)
+		signer := types.LatestSigner(gspec.Config)
+		gasLimit := uint64(1e7)
+		setupId := crypto.Keccak256([]byte("setUp()"))[:4]
 
-	_, _, receipts := core.GenerateChainWithGenesisWithConcrete(gspec, ethash.NewFaker(), 1, concreteRegistry, func(ii int, block *core.BlockGen) {
-		for _, id := range [][]byte{setupId, method.ID} {
-			tx := types.NewTransaction(block.TxNonce(senderAddress), contractAddress, common.Big0, gasLimit, block.BaseFee(), id)
-			signed, err := types.SignTx(tx, signer, key)
-			if err != nil {
-				panic(err)
+		_, _, receipts := core.GenerateChainWithGenesisWithConcrete(gspec, ethash.NewFaker(), 1, concreteRegistry, func(ii int, block *core.BlockGen) {
+			for _, id := range [][]byte{setupId, method.ID} {
+				tx := types.NewTransaction(block.TxNonce(senderAddress), contractAddress, common.Big0, gasLimit, block.BaseFee(), id)
+				signed, err := types.SignTx(tx, signer, key)
+				if err != nil {
+					t.Fatal(err)
+				}
+				block.AddTx(signed)
 			}
-			block.AddTx(signed)
-		}
-	})
+		})
 
-	if len(receipts[0]) != 2 {
-		return 0, fmt.Errorf("expected 2 receipts, got %d", len(receipts))
-	}
-	setupReceipt := receipts[0][0]
-	testReceipt := receipts[0][1]
-	if setupReceipt.Status != types.ReceiptStatusSuccessful {
-		return 0, fmt.Errorf("setup failed")
-	}
-	if (testReceipt.Status == types.ReceiptStatusSuccessful) == shouldFail {
-		return 0, fmt.Errorf("test failed")
-	}
+		t.Run("validateReceipts", func(t *testing.T) {
```

Comment:
Again, why start a subtest here? Each method corresponds to a test, we are just validating receipts here, not testing anything new.